keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -18,17 +19,26 @@ interface ArtifactRepository {
    * @return `true` if a new version is persisted, `false` if the specified version was already
    * known (in which case this method is a no-op).
    */
-  fun store(artifact: DeliveryArtifact, version: String): Boolean
+  fun store(artifact: DeliveryArtifact, version: String, status: ArtifactStatus): Boolean
 
-  fun versions(artifact: DeliveryArtifact): List<String>
+  /**
+   * @returns the versions we have for an artifact, optionally filtering by status if provided
+   */
+  fun versions(
+    artifact: DeliveryArtifact,
+    statuses: List<ArtifactStatus> = enumValues<ArtifactStatus>().toList()
+  ): List<String>
+  // todo eb: if the list matches the default value, should we also include nulls? probably.
```

Comment:
Open question: should we allow `null` status to be okay if the list of statuses contains every possible value, or should we ignore nulls? This is only really relevant for the transition time going from the previous version to a new artifact being sent to us on this version. I'm inclined to just ignore this case.

Reply:
Conversely we could just make the default value be `null` and then not restrict the query.

Reply:
Do you think that's better than having the default be everything?

Reply:
In the unlikely event of a new status type being added it would continue to work!